### ISSUE TITLE
[ARCUS] hubble에서 connection 정보 보강

### ISSUE THREAD

**Issue by jhpark816** [naver/kaist-oss-course#59]
Arcus hubble(https://github.com/naver/arcus-hubble) 에서
connection 관련 정보로 아래 두 정보를 그래프로 나타내도록 한다.
현재는 전자만 표시되고 있다.
- current connection 수
- maximum connection 수

**Comment by ktomkim10** [comment]
제가 하겠습니다

**Comment by lsj1888** [comment]
@strollkim 진행해보시고, 잘 안되거나 궁금하신 내용있으면 이곳에 질문 올려주세요.

**Comment by jhpark816** [comment]
조만간 `develop` 브랜치가 준비될 예정인데요.
`develop` 브랜치 기준으로 작업하시고, pull request도 `develop` 브랜치 기준으로 보내주세요.

**Comment by MinWooJin** [comment]
arcus-hubble develop 브랜치 준비가 완료 되었습니다.
develop 브랜치 기준으로 작업하시고, pull request도 develop 브랜치 기준으로 보내주세요.

**Comment by ktomkim10** [comment]
넵 감사합니다!

**Comment by ktomkim10** [comment]
arcus빌드과정에서
[localhost] local: GOT=$(echo stat | nc localhost 2181 | grep Mode:); if [ -z "$GOT" ]; then echo "Mode: stale"; else echo $GOT; fi
zookeeper cluster not complete yet; sleeping 3 seconds
라는 오류가 뜨는데 문제가 무엇인지 잘 모르겠습니다ㅠㅠ

**Comment by MinWooJin** [comment]
@strollkim 
먼저,
`echo stat | nc localhost 2181 | grep Mode`
명령어가 쉘에서 정상 동작하는지 확인해보시기 바랍니다.
ubuntu에서 설치한 패키지에 따라 nc 명령어가 없을 수도 있습니다.

nc 명령어도 존재하고 위 명령도 정상적으로 실행이 되지만 위 현상이 동일하게 계속 된다면
전체 로그를 alsdn653@jam2in.com 으로 첨부해 주시기 바랍니다.

**Comment by ktomkim10** [comment]
@MinWooJin 

ubuntu@ubuntu:~$ nc
This is nc from the netcat-openbsd package. An alternative nc is available
in the netcat-traditional package.
usage: nc [-46bCDdhjklnrStUuvZz] [-I length] [-i interval] [-O length]
      [-P proxy_username] [-p source_port] [-q seconds] [-s source]
      [-T toskeyword] [-V rtable] [-w timeout] [-X proxy_protocol]
      [-x proxy_address[:port]] [destination] [port]

nc명령어를 확인해보니 위와같이 나오는데 그냥 로그를 보내드려야할까요?

**Comment by MinWooJin** [comment]
@strollkim 
네 arcus build dependency를 위해서 설치하신 package명, JDK, Ant version도 같이 보내주세요.

**Comment by ktomkim10** [comment]
@MinWooJin 메일 드렸습니다!

**Comment by strollkim** [comment]
@MinWooJin 

ant 실행 시에  JAVA_HOME is not defined correctly 라는 에러가 뜨는데
which java  명령어로 리턴되는 디렉토리가 JAVA_HOME이 아닌가요?

**Comment by ktomkim10** [comment]
@MinWooJin 

which java 의결과인
usr/bin/java를 bashrc파일에 JAVA_HOME으로 추가하였고
echo $JAVA_HOME을 하면
usr/bin/java가 제대로 나오는데
ant실행시에는

Error: JAVA_HOME is not defined correctly.
  We cannot execute /usr/bin/java/bin/java

이렇게 되는데 왜 /usr/bin/java/bin/java로 연결이 되는지 모르겠습니다

**Comment by MinWooJin** [comment]
@strollkim 
JAVA_HOME path를 /usr/ 까지만 하고 한번 해보세요.
해당 OS 환경설정 문제로 빌드 에러가 있었던 부분 같습니다.

**Comment by ktomkim10** [comment]
@MinWooJin  그렇게 하니까 해결되었습니다! 감사합니다!

**Comment by ktomkim10** [comment]
@MinWooJin 계속 빌드 관련 질문만 드려서 죄송합니다..ㅠ

checking for xmlParseFile in -lxml2... no
checking for pkg-config... (cached) no

## configure: WARNING:
- I could not find a working copy of libxml-2.0. Check config.log for hints on why
  this is the case. Maybe you need to set LDFLAGS and CPPFLAGS appropriately
  so that compiler and the linker can find libxml2 and its header files. If
  you have not installed libxml-2.0, you can get it either from its original home on
  
     http://xmlsoft.org/downloads.html
  
  You can find also find an archive copy on
  
     http://oss.oetiker.ch/rrdtool/pub/libs
  
  The last tested version of libxml-2.0 is 2.7.8.
  
  ```
   LIBS=-lm 
  ```
  
  LDFLAGS=
  CPPFLAGS= -I/usr/include/libxml2

---

configure: error: Please fix the library issues listed above and try again.

와 같은식으로 라이브러리가 설치되지 않았다는 에러가 계속 나는데
curl -OL 사이트 로 설치해도 없어지질 않습니다. 설치해야 하는 정해진 경로가 있나요?

**Comment by MinWooJin** [comment]
@strollkim 
아래 사이트 참고해보시고 안되시면 어떤 과정에서 에러가 났는지 알려주세요.
http://ubuntuforums.org/showthread.php?t=836133

**Comment by ktomkim10** [comment]
@MinWooJin 

ake  all-recursive
make[1]: Entering directory `/home/ubuntu/vendor/rrdtool-1.4.8'
Making all in po
make[2]: Entering directory`/home/ubuntu/vendor/rrdtool-1.4.8/po'
make[2]: Leaving directory `/home/ubuntu/vendor/rrdtool-1.4.8/po'
Making all in src
make[2]: Entering directory`/home/ubuntu/vendor/rrdtool-1.4.8/src'
  GEN    librrd.sym
  CCLD   librrd.la
  CCLD   librrd_th.la
  CCLD   rrdtool
  CCLD   rrdcached
/usr/bin/ld: rrdcached-rrd_daemon.o: undefined reference to symbol 'g_tree_lookup'
//lib/x86_64-linux-gnu/libglib-2.0.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: **\* [rrdcached] Error 1
make[2]: Leaving directory `/home/ubuntu/vendor/rrdtool-1.4.8/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory`/home/ubuntu/vendor/rrdtool-1.4.8'
make: **\* [all] Error 2
와 같은 에러가 발생하여 http://oss.oetiker.ch/rrdtool/forum.en.html#nabble-td7581688를 참고하여 고쳐보려고 시도하였는데 잘 되지 않았습니다.
.o파일이라 직접 건들기 조심스러운데 어떻게 해야할까요?

**Comment by minkikim89** [comment]
@strollkim 
사용하시는 Ubuntu 버전 좀 알 수 있을까요?

**Comment by ktomkim10** [comment]
@minkikim89  14.04 LTS입니다!

**Comment by minkikim89** [comment]
@strollkim 
12.04에서는 큰 문제가 없는데 14.04에서는 문제가 발생하네요
apt-get install libpango1.0-dev libxml2-dev
해서 의존성 해결해보시고 안되시면 다시 말씀해주세요

**Comment by ktomkim10** [comment]
@minkikim89  아예 12.04를 설치하고 하고 있는데 큰 문제 없이 되는것같네요! 감사합니다!
